src/components/ActionButton/ActionButton.tsx

```diff
@@ -1,24 +1,36 @@
 import classNames from "classnames";
 import PropTypes from "prop-types";
 import React, { useEffect, useRef, useState } from "react";
+import type { HTMLAttributes, ReactNode } from "react";
 
 import "./ActionButton.scss";
 
 export const LOADER_MIN_DURATION = 400; // minimium duration (ms) loader displays
 export const SUCCESS_DURATION = 2000; // duration (ms) success tick is displayed
 
+type ButtonAppearance = "base" | "neutral" | "positive" | "negative" | "brand";
+type Props = {
+  children: ReactNode;
+  appearance?: ButtonAppearance;
+  className?: string;
+  disabled?: boolean;
+  inline?: boolean;
+  loading?: boolean;
+  success?: boolean;
+} & HTMLAttributes<HTMLButtonElement>;
```

Comment:
I wonder if this should extend our Button component type instead (possibly with an Omit if anything doesn't apply).

Reply:
Good suggestion, but as `ActionButton` doesn't currently import `Button`, will defer this until later - have opened this [issue](https://github.com/canonical-web-and-design/react-components/issues/189).